v4/export/sql_type.go

```diff
@@ -101,6 +90,66 @@ func escape(s []byte, bf *bytes.Buffer, escapeQuotation []byte) {
 	}
 }
 
+func escapeBackslashCSV(s []byte, bf *bytes.Buffer, opt *csvOption) {
+	var (
+		escape  byte
+		last         = 0
+		specCmt byte = 0
+	)
+	if len(opt.delimiter) > 0 {
+		specCmt = opt.delimiter[0] // if csv has a delimiter, we should use backslash to comment the delimiter in field value
+	} else if len(opt.separator) > 0 {
+		specCmt = opt.separator[0] // if csv's delimiter is "", we should escape the separator to avoid error
+	}
+
+	for i := 0; i < len(s); i++ {
+		escape = 0
+
+		switch s[i] {
+		case 0: /* Must be escaped for 'mysql' */
+			escape = '0'
+		case '\n': /* escaped for line terminators */
+			escape = 'n'
+		case '\\':
+			escape = '\\'
+		case specCmt:
+			escape = specCmt
+		}
+
+		if escape != 0 {
+			bf.Write(s[last:i])
+			bf.WriteByte('\\')
+			bf.WriteByte(escape)
+			last = i + 1
+		}
+	}
+	if last == 0 {
+		bf.Write(s)
+	} else if last < len(s) {
+		bf.Write(s[last:])
+	}
```

Comment:
```suggestion
	bf.Write(s[last:])
```

all branches are equivalent since `s[0:] == s` and `s[len(s):] == ""`